tools/challenge-md-parser/fix-challenges.js

```diff
@@ -0,0 +1,367 @@
+  return async text => {
+    if (!text) {
+      return text;
+    }
+    try {
+      const translate = new Translate({ projectId });
+      const result = await translate.translate(text, { from, to });
+      const translations = result[0];
+      return translations;
+    } catch (err) {
+      // console.error(err);
+      return text;
+    }
+  };
+}
+
+async function processTests(tests, referenceTests, translateText) {
+  const testsObject = YAML.load(referenceTests);
+  if (
+    !testsObject.tests ||
+    testsObject.tests.length === 0 ||
+    !testsObject.tests[0].text
+  ) {
+    return referenceTests;
+  }
+  const newTests = await Promise.all(
+    testsObject.tests.map(async test => {
+      const text = await translateText(test.text);
+      return { ...test, text };
+    })
+  );
+  const testStrings = newTests
+    .map(
+      ({ text, testString }) =>
+        `  - text: ${dumpToYamlString(text)}    testString: ${dumpToYamlString(
+          testString
+        )}`
+    )
+    .join('');
+  return `tests:${testStrings ? '\n' + testStrings : ' []\n'}`;
+}
+
+function dumpToYamlString(text) {
+  let fromYaml;
+  try {
+    fromYaml = YAML.load(text);
+  } catch {
+    // console.error(`YAML load: ${text}`);
+  }
+  if (text === fromYaml) {
+    return text + '\n';
+  }
+  return YAML.dump(text, { lineWidth: 10000 });
+}
+
+function replaceWithReferenceData() {
+  return async(tree, file) => {
+    const { referenceChallenge, lang, translateText } = file.data;
+
+    if (lang && lang !== 'english') {
+      const data = YAML.load(file.data.frontmatter);
+      if (!data.localeTitle) {
+        data.localeTitle = await translateText(data.title);
+        file.data.frontmatter = YAML.dump(data, {
+          lineWidth: 10000
+        }).trimRight();
+      }
+    }
+
+    if (referenceChallenge) {
+      const { description, instructions } = file.data;
+
+      if (!description || description === 'undefined') {
+        file.data.description = await translateText(
+          referenceChallenge.description
+        );
+      }
+      if (!instructions || instructions === 'undefined') {
+        file.data.instructions = await translateText(
+          referenceChallenge.instructions
+        );
+      }
+      file.data.tests = await processTests(
+        file.data.tests,
+        referenceChallenge.tests,
+        translateText
+      );
+      file.data.files = referenceChallenge.files;
+      file.data.solutions = referenceChallenge.solutions;
+    }
+  };
+}
+
+function output() {
+  this.Compiler = function(tree, file) {
+    let {
+      frontmatter,
+      description,
+      instructions,
+      tests,
+      files: [challengeFile = {}]
+    } = file.data;
+    const { ext, contents, head, tail } = challengeFile;
+    let { solutions = [] } = file.data;
+    solutions = solutions
+      .map(s => s.trim())
+      .map(s =>
+        !s.includes('\n') && (/^\/\//).test(s) ? '// solution required' : s
+      );
```

Comment:
Also needs a replacer on the common instructions.
Eg. for Spanish:
add your code below this line -> escribe tu código debajo de este comentario
add your code above this line -> escribe tu código arriba de este comentario

Reply:
@lynxlynxlynx I had already started a side project to find all the current comments used in the English version of the challenges and see if we can condense that list down, so we could maybe have a lookup table to have the translations always be the same.  If this works, we could theoretically remove the solution section out of the non-English challenge files and just import them in from the English files and use the lookup table to make the comment translations.

Reply:
The first part is why I added the examples. Removal is a different matter, something we have to decide on, as people don't just translate comments in the seed, they also translate flavour text like "cat". The solution however doesn't matter, since it's only  for our own tests.